### PR TITLE
Delete User ,Category And Product

### PR DESCRIPTION
**Pull request by Khalid-Abed**
Create Dynamic Ajax Function To Delete  User , Category and Product
Add User Avatars Directory And including some Avatar That I Using To Test Code
Create All Categories Page To View

### CODE REVIEW

File: views/admin/product/all_products.php [hunk 1]
```diff
@@ -8,40 +8,38 @@
...
-     <button class='btn btn-primary'>Edit</button>" ."</td>".
-     "</tr> </tbody>";
-     }
-     echo "</table>";
-             
+          foreach($products as $product){
+            echo "<tr id='".$product['product_id']."'><th scope='row'>".$product['product_id']."</th>".              
+            "<td> ".$product['name']."</td>".
+            "<td> ".$product['price']."</td>".
+            "<td> ".$product['category_id']."</td>".
```
Comment (Hossam-Tarek) [inline views/admin/product/all_products.php]: You must not show data that was entered by the user to the screen before sanitizing it.

File: templates/footer.php [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <script src="/cafeteria/js/jquery.js"></script>
 <script src="/cafeteria/js/bootstrap.bundle.js"></script>
+<script src="/cafeteria/js/admin/main.js"></script>
```
Comment (Hossam-Tarek) [inline templates/footer.php]: You should not add the script file here add it in your page using the `$PAGE_SCRIPTS` variable.

File: views/admin/category/allcategories.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                echo "<div class='row'><div class='col-md-8 offset-2'>";
+                    echo "<table class='table'>
+                    <tr class='bg-dark text-light'>
+                        <th scope='col'>ID</th>
+                        <th scope='col'>Category Name</th>
+                        <th scope='col'>Action</th>
+                    </tr>";
+            foreach($categories as $category){
+                echo "<tr id='".$category['category_id']."'><th scope='row'>".$category['category_id']."</th>".              
+                "<td> ".$category['name']."</td>".
```
Comment (Hossam-Tarek) [inline views/admin/category/allcategories.php]: You must not show data that was entered by the user to the screen before sanitizing it.

File: views/admin/user/all_users.php [hunk 4]
```diff
@@ -15,44 +15,39 @@
...
-     <button class='btn btn-primary'>Edit</button>" ."</td>".
-     
-     "</tr> </tbody>";
-     }
+         foreach($users as $user){
+          echo "<tr id='".$user['user_id']."'><th scope='row'>".$user['user_id']."</th>".              
+          "<td> ".$user['name']."</td>".
+          "<td> ".$user['room_id']."</td>".
+          "<td> ".$user['email']."</td>".
+          "<td> ".$user['extra_info']."</td>".
```
Comment (Hossam-Tarek) [inline views/admin/user/all_users.php]: You must not show data that was entered by the user to the screen before sanitizing it.

**Review by Hossam-Tarek — CHANGES_REQUESTED** [review]

File: views/admin/product/all_products.php (hunk 1 above)
Comment (Khalid-Abed) [reply views/admin/product/all_products.php]: Thanks for Your Notes , But Viewing Data is Assigned to Ahmed El Hadary Not To me.

**Review by Hossam-Tarek — APPROVED** [review]

**Pull request merged**